src/knock.ts

```diff
@@ -153,7 +155,7 @@ class Knock {
   }
 
   handleErrorResponse(path: string, error: any) {
-    if (axios.isAxiosError(error) && error.response) {
+    if (error.response) {
```

Comment:
Just want to double check - will this handler still get called for non-2xx responses? Asking because I'm not sure if fetch automatically handles non-2xx responses by throwing therefore catching as an error in this.get/post/put/delete methods. It [sounds like](https://web.dev/articles/fetch-api-error-handling#when_the_network_status_code_represents_an_error) that may not be the case, hence wanting to double check.

)

Reply:
Fetch doesn't, so if we're relying on that behavior from axios, we'll need to add that back in. I'll address that in my next commit.

Reply:
Updated to throw on non-2xx errors by checking `response.ok`. Tested it with a 401 error by providing an invalid key and the error flows through correctly still.